pkg/engine/engine.go

```diff
@@ -53,7 +53,16 @@ type CallResult struct {
 	Result string `json:"result,omitempty"`
 }
 
+type commonContext struct {
+	ID           string         `json:"id"`
+	Tool         types.Tool     `json:"tool"`
+	InputContext []InputContext `json:"inputContext"`
+	// IsCredential indicates that the current call is for a credential tool
+	IsCredential bool `json:"isCredential"`
+}
+
 type Context struct {
```

Comment:
Is there any need to retain the `ID`, `Tool`, `InputContext`, and `IsCredential` fields on this main Context type, now that they are in the embedded struct commonContext?